pkg/init/asker/asker.go

```diff
@@ -77,3 +78,121 @@ func (o *Survey) AskName(defaultName string) (string, error) {
 	}
 	return answer, nil
 }
+
+// AskPersonalizeConfiguration asks the configuration user wants to change
+func (o *Survey) AskPersonalizeConfiguration(configuration ContainerConfiguration) (ContainerMap, error) {
+	options := []string{
+		"NOTHING - configuration is correct",
+	}
+	tracker := []ContainerMap{{Ops: "Nothing"}}
+	for _, port := range configuration.Ports {
+		options = append(options, fmt.Sprintf("Delete port %q", port))
+		tracker = append(tracker, ContainerMap{
+			Ops:  "Delete",
+			Kind: "Port",
+			Key:  port,
+		})
+	}
+	options = append(options, "Add new port")
+	tracker = append(tracker, ContainerMap{
+		Ops:  "Add",
+		Kind: "Port",
+	})
+
+	for key := range configuration.Envs {
+		options = append(options, fmt.Sprintf("Delete environment variable %q", key))
+		tracker = append(tracker, ContainerMap{
+			Ops:  "Delete",
+			Kind: "EnvVar",
+			Key:  key,
+		})
+	}
+	options = append(options, "Add new environment variable")
+	tracker = append(tracker, ContainerMap{
+		Ops:  "Add",
+		Kind: "EnvVar",
+	})
+
+	configChangeQuestion := &survey.Select{
+		Message: "What configuration do you want change?",
+		Default: options[0],
+		Options: options,
+	}
+	var configChangeIndex int
+	err := survey.AskOne(configChangeQuestion, &configChangeIndex)
+	if err != nil {
+		return ContainerMap{}, err
+	}
+	return tracker[configChangeIndex], nil
+}
+
+// AskAddEnvVar asks the key and value for env var
+func (o *Survey) AskAddEnvVar() (string, string, error) {
+	newEnvNameQuesion := &survey.Input{
+		Message: "Enter new environment variable name:",
+	}
+	var newEnvNameAnswer string
+	err := survey.AskOne(newEnvNameQuesion, &newEnvNameAnswer)
+	if err != nil {
+		return "", "", err
+	}
+	newEnvValueQuestion := &survey.Input{
+		Message: fmt.Sprintf("Enter value for %q environment variable:", newEnvNameAnswer),
+	}
+	var newEnvValueAnswer string
+	err = survey.AskOne(newEnvValueQuestion, &newEnvValueAnswer)
+	if err != nil {
+		return "", "", err
+	}
+	return newEnvNameAnswer, newEnvValueAnswer, nil
+}
+
+// AskAddPort asks the container name and port that user wants to add
+func (o *Survey) AskAddPort() (string, error) {
+	newPortQuestion := &survey.Input{
+		Message: "Enter port number:",
+	}
+	var newPortAnswer string
+	log.Warning("Please ensure that you do not add a duplicate port number")
+	err := survey.AskOne(newPortQuestion, &newPortAnswer)
+	if err != nil {
+		return "", err
+	}
+	return newPortAnswer, nil
+}
+
+func (o *Survey) AskContainerName(containers []string) (string, error) {
+	selectContainerQuestion := &survey.Select{
+		Message: "Select container for which you want to change configuration?",
+		Default: containers[len(containers)-1],
+		Options: containers,
+	}
+	var selectContainerAnswer string
+	err := survey.AskOne(selectContainerQuestion, &selectContainerAnswer)
+	if err != nil {
+		return selectContainerAnswer, err
+	}
+	return selectContainerAnswer, nil
+}
+
+type ContainerConfiguration struct {
```

Comment:
It would be helpful to have these types definitions into the `asker/interface.go` file, as they are part of the interface definition, and to document them, so the documentation reader can understand how data is exchanged with the methods.